server/app/repository/ProgramRepository.java

```diff
@@ -123,6 +126,35 @@ public ImmutableSet<String> getAllProgramNames() {
     return names.build();
   }
 
+  /**
+   * Gets the program definition that contains the related question data from the cache (if
+   * enabled).
+   */
+  public Optional<ProgramDefinition> getFullProgramDefinitionFromCache(ProgramModel program) {
+    if (settingsManifest.getQuestionCacheEnabled()) {
+      return programDefCache.get(String.valueOf(program.id));
```

Comment:
Question (non-blocking): I'm assuming the cache only supports lookups with strings, not with longs, which is why you need the `String.valueOf`?

Reply:
Yeah, exactly! https://www.playframework.com/documentation/2.6.x/api/java/play/cache/SyncCacheApi.html